compat/getnameinfo.cc

```diff
@@ -260,87 +252,45 @@ xgetnameinfo(const struct sockaddr *sa, socklen_t salen, char *host, size_t host
 
         goto numeric;
     } else {
-#if USE_GETIPNODEBY
-        int h_error = 0;
-        hp = getipnodebyaddr(addr, afd->a_addrlen, afd->a_af, &h_error);
-#else
         hp = gethostbyaddr(addr, afd->a_addrlen, afd->a_af);
-#if 0 // getnameinfo.c:161:9: error: variable 'h_error' set but not used
-#if HAVE_H_ERRNO
-        h_error = h_errno;
-#else
-        h_error = EINVAL;
-#endif
-#endif /* 0 */
-#endif
 
         if (hp) {
-#if 0
-            if (flags & NI_NOFQDN) {
-                /*
-                 * According to RFC3493 section 6.2, NI_NOFQDN
-                 * means "node name portion of the FQDN shall
-                 * be returned for local hosts."  The following
-                 * code tries to implement it by returning the
-                 * first label (the part before the first
-                 * period) of the FQDN.  However, it is not
-                 * clear if this always makes sense, since the
-                 * given address may be outside of "local
-                 * hosts."  Due to the unclear description, we
-                 * disable the code in this implementation.
-                 */
-                char *p;
-                p = strchr(hp->h_name, '.');
-                if (p)
-                    *p = '\0';
-            }
-#endif
             if (strlen(hp->h_name) + 1 > hostlen) {
-#if USE_GETIPNODEBY
-                freehostent(hp);
-#endif
                 return EAI_OVERFLOW;
             }
             xstrncpy(host, hp->h_name, hostlen);
-#if USE_GETIPNODEBY
-            freehostent(hp);
-#endif
         } else {
             if (flags & NI_NAMEREQD)
                 return EAI_NONAME;
 
-numeric:
+        numeric:
             switch (afd->a_af) {
 #if INET6
-            case AF_INET6: {
+            case AF_INET6:
                 int error;
```

Comment:
Removing these curly braces will lead to compile errors if this code is ever compiled: https://stackoverflow.com/questions/92396/why-cant-variables-be-declared-in-a-switch-statement

Please note that I the `if INET6` guard is probably a bug. Fetchmail defines that macro and Squid does not:

```C
configure.ac:    AC_DEFINE(INET6,1,Define to 1 if your system defines AF_INET6 and PF_INET6.)
```

This problem highlights why it is usually a bad idea to polish code without proper testing.